controllers/wavefrontalert_controller.go

```diff
@@ -335,3 +339,57 @@ func (r *WavefrontAlertReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		WithEventFilter(controllercommon.StatusUpdatePredicate{}).
 		Complete(r)
 }
+
+func (r *WavefrontAlertReconciler) UpdateIndividualWavefrontAlertStatusError(
+	ctx context.Context,
+	wfAlert *alertmanagerv1alpha1.WavefrontAlert,
+	state alertmanagerv1alpha1.State,
+	err error,
+	requeueTime ...float64,
+) (ctrl.Result, error) {
+	log := log.Logger(ctx, "controllers", "wavefrontalert_controller", "UpdateIndividualWavefrontAlertError")
+	log = log.WithValues("wavefront_cr", wfAlert.Name, "namespace", wfAlert.Namespace)
+
+	wfAlert.Status.RetryCount = wfAlert.Status.RetryCount + 1
+	wfAlert.Status.ErrorDescription = err.Error()
+	wfAlert.Status.State = state
+
+	log.Error(err, "error occurred in wavefront alert", "wavefrontAlert", wfAlert.Name)
+	r.Recorder.Event(wfAlert, v1.EventTypeWarning, err.Error(), fmt.Sprintf("error occurred in wavefront alert %s", wfAlert.Name))
+	return r.CommonClient.UpdateStatus(ctx, wfAlert, state, requeueTime[0])
+}
+
+func (r *WavefrontAlertReconciler) UpdateIndividualWavefrontAlert(
+	ctx context.Context,
+	req ctrl.Request,
+	alertStatus alertmanagerv1alpha1.AlertStatus,
+	wfAlert alertmanagerv1alpha1.WavefrontAlert,
+) error {
+	log := log.Logger(ctx, "controllers", "wavefrontalert_controller", "UpdateIndividualWavefrontAlert")
+	// Get alertsconfig CR
+	alertConfigNamespacedName := types.NamespacedName{Namespace: req.Namespace, Name: alertStatus.AssociatedAlertsConfig.CR}
+	var alertsConfig alertmanagerv1alpha1.AlertsConfig
+	if err := r.Get(ctx, alertConfigNamespacedName, &alertsConfig); err != nil {
+		return err
+	}
+	// Get the corresponding alert in alertsConfig
+	config := alertsConfig.Spec.Alerts[alertStatus.AssociatedAlert.CR]
+	var alert wf.Alert
+	// Create wavefront alert with proper substituted value of that exported param
+	if err := controllercommon.GetProcessedWFAlert(ctx, &wfAlert, &config, &alert); err != nil {
+		return err
+	}
+	// Update alert in wavefront
+	alert.ID = &alertStatus.ID
+	// Validate the alert request
+	if err := wavefront.ValidateAlertInput(ctx, &alert); err != nil {
+		return err
+	}
+
+	if err := r.WavefrontClient.UpdateAlert(ctx, &alert); err != nil {
+		return err
+	}
+	// We do not need to update individual alertStatus here because there is no change in alertsConfig
+	log.Info("alert successfully got updated", "alertID", alert.ID)
```

Comment:
Just FYI.. Its not included now but we probably have to do "LastUpdatedTimeStamp" in separate PR and we may have to update timestamp at that time

Reply:
Also, we may need to introduce the version in AlertStatus so we know what was the last version it was updated too. It could be as simple as "Generation" number so in future by looking at the status we know what was the last version it was applied to.

Reply:
sounds good, can you create separate issue for those things? thanks

Reply:
Anything other comments to update the PR? if not, please approve it, thanks!

Reply:
https://github.com/keikoproj/alert-manager/pull/21#pullrequestreview-748547826 - Does that testing went through-?

Reply:
Are you talking about updating alertsconfig with severity warning? that goes through alertsconfig controller reconcile logic and errored/reties from there. It does not call my code. If you want to test error scenaior in wavefrontalert, I need to figure out a way to modify wavefrontalert to error out, let me try that. But I do see the code needs to update alertStatus in alertsconfig for error case, will add that and test it out

Reply:
Updated PR with error handling and please check out my "Test case 3: error handling" in PR description

Reply:
Hmm. I'm trying to NOT to update other CR if there is an error. For example: If one of the alertsConfig did not provide one of the exportParam- error should be on alertsConfig side and not WavefrontAlert. But, Its too confused for now and i think that will change eventually and once we have better picture we can take that decision

Reply:
And also, I was just asking whether error scenario is getting retried and looks like it IS based on your test results where retry count is 5 or something.

Reply:
We will proceed for now and we will revisit the idea of updating the status in error scenario again

Reply:
I am updating the alert status for error/good scenario out of this function, I will remove the old comments

Reply:
We can revisit the idea of updating the status in error scenario, but I think we should keep the alert status on both CR consistent otherwise it is confusing since they are pointing to the same alert, thanks